server/src/main/java/com/cloud/user/DomainManagerImpl.java

```diff
@@ -220,11 +220,10 @@ public DomainVO doInTransaction(TransactionStatus status) {
                 _resourceCountDao.createResourceCounts(domain.getId(), ResourceLimit.ResourceOwnerType.Domain);
 
                 CallContext.current().putContextParameter(Domain.class, domain.getUuid());
-                _messageBus.publish(_name, MESSAGE_ADD_DOMAIN_EVENT, PublishScope.LOCAL, domain.getId());
                 return domain;
             }
         });
-
+        _messageBus.publish(_name, MESSAGE_ADD_DOMAIN_EVENT, PublishScope.LOCAL, domain.getId());
```

Comment:
Just wondering if this is cause for concern, what if the transaction fails for some reason @Pearl1594, should we check for that (for ex. the domain ID is not null) ?

Reply:
Thanks @rohityadavcloud . I've added a check